### ISSUE TITLE
table Filters 

### ISSUE THREAD

**Issue by yagogak** [laravel-filament/filament#10819]
### Package

filament/filament

### Package Version

v3.2.0

### Laravel Version

v10.10

### Livewire Version

v3.0.0

### PHP Version

PHP 8.1.0

### Problem description

Just upgrading from 3.0-stable to 3.2.0 I got multiple toogle filters on a table. The data didn't refresh after the first click on the toogle, i need to click a second toogle for the first to be applied ( or refresh the page )

```
public static function getFilters(): array
    {
        $filters = [];

        // get all possible metadatas keys for Sessions
        $metadatasKeys = Session::all()->map(function ($session) {
            return collect($session->metadatas)->keys();
        })->flatten()->unique();

        foreach ($metadatasKeys as $metadatasKey) {
            $fi = Filter::make('hasMetadatas'.$metadatasKey)
                ->label('metadatas '.$metadatasKey)
                ->query(fn ($query) => $query->ByMetadataKeyExist($metadatasKey));
            $filters[] = $fi;
        }

        return $filters;
    }

```

### Expected behavior

those filters used to work properly in 3.0-stable

### Steps to reproduce

place multiple toogle filters on the top of a table. click on the first, nothing append, click on the second, dataset is refresh but only with the first filter.

### Reproduction repository

https://github.com/yagogak/filamentphp-bug-filter

**Comment by saulens22** [comment]
It may be similar to a bug I found. Here's a copy from Discord:

We are using DatePicker in filter like this:
```php
Filter::make('filter')
    ->form([
        DatePicker::make('date-filter')
            ->native(false),

        TextInput::make('text-filter'),
    ]);
```
After updating to 3.2.0 it breaks.
If I update date filter - nothing happens. When I select text field and leave it - update triggers.

v3.2.0 changed filters here: https://github.com/filamentphp/filament/blob/a1fe2b159e94c9e0a6bf7405001eff074e3df4eb/packages/tables/src/Concerns/HasFilters.php#L35

In my specific scenario I tried adding live() on DatePicker, but it also happens when using toggle(), where I'm not in control of form elements.

Reproduction: https://github.com/filamentphp/demo/compare/main...saulens22:filament-demo:3.2.0-break
